lib/full-site-editing/template-parts.php

```diff
@@ -118,3 +147,38 @@ function set_unique_slug_on_create_template_part( $post_id ) {
 	}
 }
 add_action( 'save_post_wp_template_part', 'set_unique_slug_on_create_template_part' );
+
+/**
+ * Returns a filtered list of allowed section types for template parts.
+ *
+ * @return array The supported template part section types.
+ */
+function gutenberg_get_allowed_template_part_section_types() {
+	$default_section_types = array(
+		'header',
```

Comment:
What do you think about introducing these as constants? It might be beneficial down the road for comparisons, autocomplete, avoiding bugs due to typos etc.

Reply:
Im not sure I follow exactly what you are suggesting, but it sounds like it would make sense.

Reply:
I'm thinking something along the lines of:

define( `TEMPLATE_PART_SECTION_HEADER', 'header' );

and then using that throughout the code instead. It probably needs to be prefixed with `gutenberg` too, at which points it becomes somewhat unwieldy. Which makes me think that we could isolate this code within `WP_Template_Part` class to avoid the need for prefixing functions and constants all the time.

Reply:
defined them as constants here - d39b70f.  Im not sure if its necessary to check `defined(` first, but it seems like it couldn't hurt.